### PR TITLE
[AzureMonitorExporter] Propagate node attributes

### PR DESCRIPTION
**Pull request by pcwiese**
These are attributes that apply to any Span and so should be propagated.
Specifically set the ai.cloud.name and ai.cloud.instance tags on the
envelope to provide a better Azure Monitor Application Map experience.

### CODE REVIEW

**Review by pjanotti — APPROVED** [review]
LGTM

File: exporter/azuremonitorexporter/traceexporter.go [hunk 1]
```diff
@@ -653,6 +658,39 @@ func (exporter *traceExporter) pushTraceData(
...
 		if envelope, err := exporter.spanToEnvelope(exporter.config.InstrumentationKey, wireFormatSpan); err == nil && exporter.transportChannel != nil {
+
+			// Attach node level attributes to envelope and data as appropriate
+			if traceData.Node != nil && traceData.Node.Attributes != nil {
+				// Augment the envelope and envelope data with node level attributes
+				// Configure the ai.cloud.role and ai.cloud.roleinstance on the envelope tags itself, then copy the node attributes to the envelope data as well.
+				// Assumes the node level attribute key/values correspond to:
+				// https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/data-resource-semantic-conventions.md
+				if serviceName, ok := traceData.Node.Attributes[resourceAttributeKeyServiceName]; ok {
+					envelope.Tags[contracts.CloudRole] = serviceName
```
Comment (SergeyKanzhelev) [inline exporter/azuremonitorexporter/traceexporter.go]: perhaps also prepend `service.namespace` to is.

**Review by SergeyKanzhelev — APPROVED** [review]

File: exporter/azuremonitorexporter/traceexporter.go (hunk 1 above)
Comment (pcwiese) [reply exporter/azuremonitorexporter/traceexporter.go]: good idea.

File: exporter/azuremonitorexporter/traceexporter.go [hunk 2]
```diff
@@ -71,6 +71,12 @@ const (
 	spanAttributeKeyDbInstance  = "db.instance"
 	spanAttributeKeyDbStatement = "db.statement"
 	spanAttributeKeyDbUser      = "db.user"
+
+	// Resource keys
+	// https://github.com/open-telemetry/opentelemetry-specification/blob/master/specification/data-resource-semantic-conventions.md
+	resourceAttributeKeyServiceName       = "service.name"
+	resourceAttributeKeyServiceNamespace  = "service.namespace"
+	resourceAttributeKeyServiceInstanceID = "service.instance.id"
```
Comment (tigrannajaryan) [inline exporter/azuremonitorexporter/traceexporter.go]: These are already defined in https://github.com/open-telemetry/opentelemetry-collector/blob/master/translator/conventions/opentelemetry.go
Any reason for redefining and not importing?

Comment (pcwiese) [reply exporter/azuremonitorexporter/traceexporter.go]: No, I'll definitely import.

**Comment by pcwiese** [comment]
any other comments here?

**Review by tigrannajaryan — APPROVED** [review]

File: exporter/azuremonitorexporter/go.mod [hunk 3]
```diff
@@ -10,7 +10,7 @@ require (
 	github.com/census-instrumentation/opencensus-proto v0.2.1
 	github.com/golang/groupcache v0.0.0-20191027212112-611e8accdfc9 // indirect
 	github.com/golang/protobuf v1.3.2
-	github.com/open-telemetry/opentelemetry-collector v0.2.1-0.20191016224815-dfabfb0c1d1e
+	github.com/open-telemetry/opentelemetry-collector v0.2.1-0.20191217213608-3cac55bf869f
```
Comment (tigrannajaryan) [inline exporter/azuremonitorexporter/go.mod]: Please rebase and revert this change. Collector is now at 0.2.2.
